### PR TITLE
Initial ginkgo framework

### PR DESCRIPTION
**Pull request by baude**
This is an initial attempt at bringing in the ginkgo test framework into
buildah.  The inspect bats file was also imported.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: tests/e2e/buildah_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+	"github.com/containers/storage/pkg/reexec"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/buildah"
+)
+
+var (
+	PODMAN_BINARY      string
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: I'm not sure we'll need this.  Do you foresee a use for it?

File: tests/e2e/buildah_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+	CNI_CONFIG_DIR     string
+	RUNC_BINARY        string
+	INTEGRATION_ROOT   string
+	STORAGE_OPTIONS    = "--storage-driver vfs"
+	ARTIFACT_DIR       = "/tmp/.artifacts"
+	CACHE_IMAGES       = []string{"alpine", "busybox", fedoraMinimal}
+	RESTORE_IMAGES     = []string{"alpine", "busybox"}
+	ALPINE             = "docker.io/library/alpine:latest"
+	BB_GLIBC           = "docker.io/library/busybox:glibc"
+	fedoraMinimal      = "registry.fedoraproject.org/fedora-minimal:latest"
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: FEDORA_MINIMAL?  or was there a reason to go camelback with this?

File: tests/e2e/buildah_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+	defaultWaitTimeout = 90
+)
+
+// BuildAhSession wrapps the gexec.session so we can extend it
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: nit 'wrapps' to 'wraps'

File: tests/e2e/buildah_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+type BuildAhSession struct {
+	*gexec.Session
+}
+
+// BuildAhTest struct for command line options
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: OK, I'm giggling everytime I see BuildaAh*, nicely done sir!

File: tests/e2e/buildah_suite_test.go [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+	RegistriesConf string
+}
+
+// TestBuildAh ginkgo master function
+func TestBuildAh(t *testing.T) {
+	if reexec.Init() {
+		os.Exit(1)
+	}
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Libpod Suite")
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: Buildah suite?

File: tests/e2e/buildah_suite_test.go [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+})
+
+// CreateTempDirin
+func CreateTempDirInTempDir() (string, error) {
+	return ioutil.TempDir("", "buildah_test")
+}
+
+// PodmanCreate creates a BuildAhTest instance for the tests
+func PodmanCreate(tempDir string) BuildAhTest {
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: BuildahCreate perhaps?

File: tests/e2e/buildah_suite_test.go [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+	}
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Libpod Suite")
+}
+
+var _ = BeforeSuite(func() {
+	//Cache images
+	cwd, _ := os.Getwd()
+	INTEGRATION_ROOT = filepath.Join(cwd, "../../")
+	podman := PodmanCreate("/tmp")
```
Comment (TomSweeneyRedHat) [inline tests/e2e/buildah_suite_test.go]: podman to buildah?

**Comment by TomSweeneyRedHat** [comment]
@baude looks like .papr.yml might need some love. Red Hat CI can't find gingko

File: tests/e2e/buildah_suite_test.go (hunk 1 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: Nope, removed

File: tests/e2e/buildah_suite_test.go (hunk 2 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: no reason, fixed

File: tests/e2e/buildah_suite_test.go (hunk 3 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: fixed

File: tests/e2e/buildah_suite_test.go (hunk 5 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: fixed

File: tests/e2e/buildah_suite_test.go (hunk 6 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: fixed

File: tests/e2e/buildah_suite_test.go (hunk 7 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: fixed

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@nalind WDYT?

**Comment by TomSweeneyRedHat** [comment]
Green buttons, @nalind can you take a looksee please?

File: tests/e2e/buildah_suite_test.go (hunk 4 above)
Comment (nalind) [reply tests/e2e/buildah_suite_test.go]: Yeah, I don't really get that.  There's a few stray references to podman in here, too?

**Comment by nalind** [comment]
Looks mostly fine.

File: tests/e2e/buildah_suite_test.go (hunk 4 above)
Comment (baude) [reply tests/e2e/buildah_suite_test.go]: ill correct the strays

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**